src/classes/PPlan_Primary_TDTM.cls

```diff
@@ -51,27 +58,50 @@ public with sharing class PPlan_Primary_TDTM extends TDTM_Runnable {
         Set<Id> primaryProgramIds = new Set<Id>();
         // BEFORE INSERT
         if ( newlist != null && triggerAction == TDTM_Runnable.Action.BeforeInsert && !TDTM_ProcessControl.getRecursionFlag(TDTM_ProcessControl.registeredTrigger.PPlan_Primary_TDTM_Before_Insert)) {
+            for (integer i=0; i<newlist.size(); i++) {
+                Program_Plan__c newPPlan = (Program_Plan__c)newlist[i];
+                if (newPPlan.Is_Primary__c == true) {
+                    // Make Plan "Current" if "Is Primary" is checked.
+					newPPlan.Status__c = STATUS_CURRENT_VAL;
+                }
+            }
+        }
+        // AFTER INSERT
+        if ( newlist != null && triggerAction == TDTM_Runnable.Action.AfterInsert && !TDTM_ProcessControl.getRecursionFlag(TDTM_ProcessControl.registeredTrigger.PPlan_Primary_TDTM_After_Insert)) {
             for (integer i=0; i<newlist.size(); i++) {
                 Program_Plan__c newPPlan = (Program_Plan__c)newlist[i];
                 if (newPPlan.Is_Primary__c == true) {
                     //if multiple program plans under one single account are set as primary in same transation
                     //Use the first program plan as the primary
-                    if (!acccountIdsNeedNonPrimarycount.contains(newPPlan.Account__c)) {
+					if (!acccountIdsNeedNonPrimarycount.contains(newPPlan.Account__c)) {
                         primaryProgramIds.add(newPPlan.Id);
                     }
                     acccountIdsNeedNonPrimarycount.add(newPPlan.Account__c);
                 }
             }
         }
-        // BEFORE UPDATE
+ 		// BEFORE UPDATE
         if ( newlist != null && triggerAction == TDTM_Runnable.Action.BeforeUpdate && !TDTM_ProcessControl.getRecursionFlag(TDTM_ProcessControl.registeredTrigger.PPlan_Primary_TDTM_Before_Update)) {
+           for (integer i=0; i<newlist.size(); i++) {
+                Program_Plan__c newPPlan = (Program_Plan__c)newlist[i];
+                Program_Plan__c oldPPlan = (Program_Plan__c)oldlist[i];
+                if (newPPlan.Is_Primary__c == true&& newPPlan.Is_Primary__c != oldPPlan.Is_Primary__c) {
```

Comment:
I know this one is nit-picky but can you add a space between 'true&&' so that it is 'true &&'